src/LegacyPaymentGateways/Adapters/LegacyPaymentGatewayAdapter.php

```diff
@@ -70,22 +68,23 @@ public function handleBeforeGateway(array $legacyDonationData, PaymentGatewayInt
                 'donorId' => $donor->id,
                 'installments' => (int)$subscriptionData->times,
                 'status' => SubscriptionStatus::PENDING(),
-                'donationFormId' => $formData->formId
+                'donationFormId' => $formData->formId,
             ]);
```

Comment:
I took a stab at resolving this in 398671c, but I'd like your input.

We're chasing our tail a bit here. It is required — in the database — so there's no getting around that. And I made the bump function to make it easy to set the `renewsAt` property. Normally, I'd say we just need to do that when building a Subscription, but this wouldn't work for `Subscription::create()`, as that immediately builds the model and saves it.

I thought about simply bumping it in the `create` function, but thought perhaps it's just useful to have it auto-generated when saving. I then did it in the repository since the properties required for that function are validated. This meant I had to remove it from the required list, so I added a manual check to `SubscriptionRepository::update()`. I _think_ this is fine, practically speaking, it just feels a little weird.

Open to your thoughts!

Reply:
This solution is fine.  Having to define the `renewsAt` property when you create a subscription would be tedious because the default relies on the frequency and period which we're already defining.  Since we're validating those properties _before_ dynamically setting the property, there should be no issues.  This is pretty much how it was working with the previous `$subscription->expiration()` method which was set under-the-hood _after_ creating the subscription.

Reply:
I wouldn't mind requiring folks to define the `renewsAt`, as they could do something like:
```php
$sub->frequency = 1;
$sub->period = SubscriptionPeriod::MONTH();
$sub->bumpRenewalDate();
```
It's the specific complication this creates with `Subscription::create()` that mostly bothers me. So the question is whether `renewsAt` is generated at that point specifically, or as a side-effect whenever saving.

Reply:
I think in this case it makes the most sense to keep what you have (as a side-effect during save) - in fact our other time properties like `createdAt` and `updatedAt` work the same way.  Unless defined, they are generated during persistence. In order to set the property dynamically during create, we would have to add additional validation for the properties (like frequency & period) it requires to be generated which is not common to any other models.